src/commands/user-data/list.ts

```diff
@@ -0,0 +1,109 @@
+import * as cosmos from "@azure/cosmos";
+import { Command } from "@oclif/command";
+import cli from "cli-ux";
+
+import { getCosmosConnection, pickAzureConfig } from "../../utils/azure";
+
+export default class UserDataProcessingRequestsList extends Command {
+  public static description = "Lists all user data processing request";
+
+  public static flags = {
+    ...cli.table.flags()
+  };
+
+  public async run(): Promise<void> {
+    const { flags } = this.parse(UserDataProcessingRequestsList);
+
+    try {
+      const config = await pickAzureConfig();
+      cli.action.start("Retrieving cosmosdb credentials");
+      const { endpoint, key } = await getCosmosConnection(
+        config.resourceGroup,
+        config.cosmosName
+      );
+      cli.action.stop();
+
+      cli.action.start("Querying user requests...");
+      const client = new cosmos.CosmosClient({ endpoint, auth: { key } });
+      const database = client.database(config.cosmosDatabaseName);
+      const container = database.container(
+        config.cosmosUserDataProcessingContainer
+      );
+      const response = container.items.query(
+        "SELECT c.fiscalCode, c.choice, c.updatedAt, c.status, c.version FROM c",
+        {
+          enableCrossPartitionQuery: true
+        }
+      );
+      const result = (await response.toArray()).result;
+
+      cli.action.stop();
+      if (result === undefined) {
+        this.error("No result");
+        return;
+      }
+
+      const latest = Object.values(
```

Comment:
```ts
      const latest = Object.values(
        // tslint:disable-next-line: no-any
        result.reduce((prev, curr: any) => {
          const isNewer =
            !prev[curr.version] || curr.version > prev[curr.fiscalCode].version;
          return {
            ...prev,
            ...(isNewer ? { [curr.fiscalCode]: curr } : {})
          };
        }, {})
      );
```

Probably issue @gunzip:
if there is a request with different choice, only last choice is reported.
So if I request a DELETE and then a DOWNLOAD operation, only DOWNLOAD is reported and DELETE is lost.

Reply:
well, good catch. can you provide a fix?

Reply:
yes, i can try it! 👍

Reply:
```ts
      const latest = Object.values(
        // tslint:disable-next-line: no-any
        result.reduce((prev, curr: any) => {
          const isNewer =
            !prev[curr.fiscalCode + "-" + curr.choice] || (curr.version > prev[curr.fiscalCode + "-" + curr.choice].version);
          return {
            ...prev,
            ...(isNewer ? { [curr.fiscalCode + "-" + curr.choice]: curr } : {})
          };
        }, {})
      );
```

possible fix: this try to use a combined key `fiscalCode + "-" + choice`

I think there was also another little issue here `const isNewer = !prev[curr.version] ||...` -> `const isNewer = !prev[curr.fiscalCode + "-" + curr.choice] || ...`